### DISCUSSION TITLE
HA adding OpenHASP device from another network

### DISCUSSION THREAD

**Discussion by wanthalf** [HASwitchPlate/openHASP#587] [Q&A]
Hi, what is the best way to add a new OpenHASP device to HA if it is located in another network and cannot be discovered automatically (I suppose that is the problem). I tried to set up the IP of the HA to the MQTT settings in the OpenHASP configuration, but it did not help. Don't know how it is supposed to work.
I suppose I could connect and configure the device in the same network and then just move it and change its IP address in the HA configuration file `core.config_entries` (as I did it with an ESPhome device), but is there no official/correct solution?

**Comment by fvanroie** [comment]
As long as both HA and the openHASP device are connected to the same MQTT server, discovery should work fine.
What are the version of openHASP firmware and openHASP Custom Component you are using?

**Comment by wanthalf** [comment]
Latest OpenHASP component version 0.7.1, the device uses latest firmware 0.7.0-rc8. I have mosquitto addon installed and configured and I can see the plate is connecting in the logs. However, I have no idea what mqtt server the OpenHASP component is using. There is no configuration and I do not remember setting any mqtt server nor username and password for the mosquitto server anywhere. The mosquitto addon only reports some zigbee2mqtt devices (I have explicitly configured the mqtt server there). What MQTT server does OpenHASP use by default and how can I configure it? The first OpenHASP device runs just OK, somehow, and it is configured to the use the same IP address and port as mqtt server.

**Reply by fvanroie** [reply]
> However, I have no idea what mqtt server the OpenHASP component is using. 

It uses the MQTT Integration in Home Assistant, as that's a prerequisite to use MQTT in HA. You configure the server there.
It is separate from the Moquitto Broker add-on.

You also need to enable _"If newly discovered devices for MQTT should be automatically added."_ in the MQTT integration.

**Comment by wanthalf** [comment]
Oh, thank you. But now I am totally confused. What is the difference between the two? Are they just two alternatives running in parallel then? Or somehow interconnected by default? Both are configured to use the port 1883 (Mosquitto also the other three ports, of course), so which one is actually exposed to the outside IP and why? (Both OpenHAS devices are configured in the same way - to use the same IP/port - but only when connected to the same LAN, they can be discovered. Why?)

Anyway, I do not see any option such as "If newly discovered devices for MQTT should be automatically added.", anywhere.

**Comment by wanthalf** [comment]
So, I found out that discovery didn't work even in the local LAN anymore. (No idea what was changed in between.) I checked the configuration of the MQTT integration again from the new point of view (thanks for the hint) and found that there was some old default setting of username and password. (And yes, now I noticed the "discovery" setting as well, but it was enabled.) If I understand it correctly, these were the credentials used by the internal MQTT system to connect to the Mosquitto broker? Anyway, correcting the credentials solved the problem. At least for the local LAN.  Now, I will try to check again whether the discovery works from the other network as well.

**Reply by fvanroie** [reply]
Only Mosqiotto is the server and exposes the ports. All other devices are clients. HA also needs to be configured as a client of the MQTT server as well. As long as the devices can connect to the broker server, it should work.

**Comment by wanthalf** [comment]
Still no success. I am thinking the problem might now be that I turned the device on in the local LAN and asked the HA to ignore the device. Now, in the other LAN, nothing appears again. I tried to revert the "ignore device" setting (because I am not sure if the IP address or the device name or some other identifier decides). Still, nothing appears anywhere. The OpenHASP device reports it is connected to the MQTT broker. I tried restarting. Sometimes it shows some MQTT "failures" at the beginning, but their counter does not rise, while the counter of "published" messages does rise all the time. So, I would expect the communication works.

**Reply by wanthalf** [reply]
Finally! After restarting both HA and the plate, it was discovered. The configuration failed, however. In the debug logs it complained about missing configuration for the new plate in configuration.yaml, so I added some and now it works. Don't know if this is the expected behaviour (I do not remember such issue with the first plate), but it works.

Thanks for clarification of the details!
